goleveldb_batch.go

```diff
@@ -76,3 +78,7 @@ func (b *goLevelDBBatch) Close() error {
 	}
 	return nil
 }
+
+func (b *goLevelDBBatch) Compact(start, end []byte) error {
```

Comment:
No, goleveldb_batch is treated as an entirely different DB so this ,when called from Comet, will trigger compaction on the underlying DB. It is not batch specific.

Reply:
Then why have this method on a batch?

Reply:
You do have a point that in practice, the way this is used (`golevelDBBatch` ) , is via a db (calling `db.NewBatch()`). I added this to Batch because in theory, Comet could have instantiated at some point a `newGoLevelDBBatch` and try to compact on that because it does not have access to the private `db` variable inside a batch. So compaction is not done on a batch, it is done on the underlying database to which you are writing the batch. 

However, looking at the interfaces now, and the way they are used, it may make sense to remove `Compact` from the `Batch` interface and keep it only in the `DB` interface.  So that people are not encouraged to create BatchDBs directly in the first place (as it seems to me that the interface design was done with that in mind).

Reply:
Removed, please verify if it's ok now.